### ISSUE TITLE
How do I set up the options to take optional values?

### ISSUE THREAD

**Issue by FrankDMartinez** [docopt/docopt.cpp#53]
I might not quite understand the documentation correctly but, suppose I want to set a certain quality to a default level if an option is not passed to the application, a different default value if the option is passed with no arguments, and a third value if the option is passed with certain arguments?

**Comment by jaredgrubb** [comment]
I dont think this is possible because I'm not sure how you would express that in documentation in a way that is understandable to the docopt algorithm. Do you have any ideas?

**Comment by FrankDMartinez** [comment]
Not immediately, no. That's why I asked.

**Comment by janwilmans** [comment]
I think #72 is talking about the same feature request and mentions an example

**Comment by isometriq** [comment]
maybe like this?
`[default: (.*), implicit: (.*)]`
(in the `Options:`  block)

**Comment by isometriq** [comment]
I would like validation too:
https://github.com/docopt/docopt.cpp/issues/86

**Comment by janwilmans** [comment]
Maybe like this ?

```
Options:
--always_set=[<n>]     This has the current behaviour  [default: 10].
--always_set                 This implicitly sets the value to 5 [value: 5]
```
